russh/src/channels/io/tx.rs

```diff
@@ -58,11 +57,13 @@ where
         self.window_size_fut.take();
 
         let writable = (self.max_packet_size)
-            .min(*window_size)
+            .min(window_size.get())
             .min(buf.len() as u32) as usize;
         if writable == 0 {
-            // TODO fix this busywait
-            cx.waker().wake_by_ref();
+            if buf.is_empty() || self.sender.is_closed() {
```

Comment:
I am not sure if `buf.is_empty()` should lead to the channel being closed, but maybe the condition above should only be reached when the window_size is zero?